backend/layers/processing/dataset_metadata_update.py

```diff
@@ -0,0 +1,288 @@
+                    pool.apply_async(
+                        self.update_h5ad,
+                        (
+                            artifact_uris[DatasetArtifactType.H5AD],
+                            current_dataset_version,
+                            new_artifact_key_prefix,
+                            new_dataset_version_id,
+                            metadata_update,
+                        ),
+                    )
+                )
+            else:
+                self.logger.error(f"Cannot find labeled H5AD artifact uri for {current_dataset_version_id}.")
+                self.update_processing_status(
+                    new_dataset_version_id, DatasetStatusKey.H5AD, DatasetConversionStatus.FAILED
+                )
+
+            if DatasetArtifactType.RDS in artifact_uris:
+                self.logger.info("Main: Starting thread for rds update")
+                artifact_jobs.append(
+                    pool.apply_async(
+                        self.update_rds,
+                        (
+                            artifact_uris[DatasetArtifactType.RDS],
+                            new_artifact_key_prefix,
+                            new_dataset_version_id,
+                            metadata_update,
+                        ),
+                    )
+                )
+            elif current_dataset_version.status.rds_status == DatasetConversionStatus.SKIPPED:
+                self.update_processing_status(
+                    new_dataset_version_id, DatasetStatusKey.RDS, DatasetConversionStatus.SKIPPED
+                )
+            else:
+                self.logger.error(
+                    f"Cannot find RDS artifact uri for {current_dataset_version_id}, and Conversion Status is not SKIPPED."
+                )
+                self.update_processing_status(
+                    new_dataset_version_id, DatasetStatusKey.RDS, DatasetConversionStatus.FAILED
+                )
+
+            if DatasetArtifactType.CXG in artifact_uris:
+                self.logger.info("Main: Starting thread for cxg update")
+                artifact_jobs.append(
+                    pool.apply_async(
+                        self.update_cxg,
+                        (
+                            artifact_uris[DatasetArtifactType.CXG],
+                            f"s3://{self.cellxgene_bucket}/{new_artifact_key_prefix}.cxg",
+                            new_dataset_version_id,
+                            metadata_update,
+                        ),
+                    )
+                )
+            else:
+                self.logger.error(f"Cannot find cxg artifact uri for {current_dataset_version_id}.")
+                self.update_processing_status(
+                    new_dataset_version_id, DatasetStatusKey.CXG, DatasetConversionStatus.FAILED
+                )
+
+        # blocking call on async jobs before checking for valid artifact statuses
+        [j.get() for j in artifact_jobs]
```

Comment:
I think you should include a comment about why we want to `.get()` and not `.wait()` here. Subtle distinction that is important which could get errantly changed in code gets shuffled around in the future.